File: php/features.php
<?php
declare(strict_types=1);

// Accommodation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AccommodationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AccommodationBaseFeature();
            case "test":
                return new AccommodationTestFeature();
            default:
                return new AccommodationBaseFeature();
        }
    }
}

File: py/features.py
# Accommodation SDK feature factory

from feature.base_feature import AccommodationBaseFeature
from feature.test_feature import AccommodationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AccommodationBaseFeature(),
        "test": lambda: AccommodationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
